src/hints/mod.rs

```diff
@@ -336,3 +339,26 @@ pub fn breakpoint(
     println!("\n-----------END BREAKPOINT-----------\n");
     Ok(())
 }
+
+#[allow(unused)]
+const IS_ON_CURVE: &str = "ids.is_on_curve = (y * y) % SECP_P == y_square_int";
+pub fn is_on_curve(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let y: BigInt = exec_scopes.get("y")?;
+    // TODO: should y_square_int be calculated or should caller put it in scope for us?
```

Comment:
Given the context, `y_square_int` is just a variable in `exec_scopes`. Looking at `cairo-lang`, I think it is set in `ec.cairo`.

Reply:
I don't see it in `ec.cairo`, can you point me to where you see that?